@Test
@DirtiesContext
public void testOnlineSell() throws Exception {
    configureMinimalGame();
    System.out.println("***** testOnlineSell *****");
    beginHttpSession();
    beginHttpRequestAndOpenSessionInViewFilter();
    movableService.sendTutorialProgress(TutorialConfig.TYPE.TUTORIAL, "", "", 0, 0);
    SimpleBase simpleBase = ((RealityInfo) movableService.getGameInfo()).getBase();
    Id id = getFirstSynItemId(simpleBase, TEST_START_BUILDER_ITEM_ID);
    clearPackets();
    movableService.sellItem(id);
    Assert.assertEquals(0, baseService.getBases().size());
    try {
        movableService.getSyncInfo();
        Assert.fail("Disconnection expected");
    } catch (NoConnectionException e) {
    // OK
    }
    // Also second call should fail
    try {
        movableService.getSyncInfo();
        Assert.fail("Disconnection expected");
    } catch (NoConnectionException e) {
    // OK
    }
    SimpleBase newBase = ((RealityInfo) movableService.getGameInfo()).getBase();
    Assert.assertEquals(1, baseService.getBases().size());
    Assert.assertFalse(newBase.equals(simpleBase));
    Message message = new Message();
    message.setMessage("You lost your base. A new base was created.");
    assertPackagesIgnoreSyncItemInfoAndClear(message);
    endHttpRequestAndOpenSessionInViewFilter();
    endHttpSession();
}
